src/middleware/addRoleContext.ts

```diff
@@ -0,0 +1,32 @@
+import type { NextFunction, Response } from 'express';
+import type { AuthenticatedRequest } from '../types';
+
+const PDC_ADMIN_ROLE = 'pdc-admin';
+
+const isRealmAccessWithRole = (obj: unknown): obj is { roles: unknown[] } =>
```

Comment:
This actually is what that's doing -- this is called a type guard: https://www.typescriptlang.org/docs/handbook/advanced-types.html

It lets you tell typescript that an object is what you expect it to be (and the reason it's safe to TELL typescript that is because you write a guard that does, in fact, confirm it to be that at runtime)